src/libraries/System.Private.CoreLib/src/System/ThrowHelper.cs

```diff
@@ -575,6 +577,34 @@ internal static void ThrowFormatIndexOutOfRange()
             throw new FormatException(SR.Format_IndexOutOfRange);
         }
 
+        [DoesNotReturn]
+        internal static void ThrowAmbiguousMatchException(MemberInfo memberInfo)
+        {
+           throw GetAmbiguousMatchException(memberInfo);
+        }
+
+        internal static AmbiguousMatchException GetAmbiguousMatchException(MemberInfo memberInfo)
```

Comment:
Made this an internal helper  because I originally thought it might be useful when we wanted an Exception to pass out (not throw), but it turns out there's only one place we do that, and it's not here... should I merge this into the method above?

Note: I copied the pattern for `GetArraySegmentCtorValidationFailedException` below, so there is precedent for this construction in this class...

Reply:
If it is not used for passing an exception probably throwing here same as above better, in case you update this method do not forget to add the `[DoesNotReturn]` attribute.